tests/acquisition/covidcast/test_csv_importer.py

```diff
@@ -0,0 +1,218 @@
+
+    self.assertTrue(CsvImporter.is_sane_week(202015))
+
+    self.assertFalse(CsvImporter.is_sane_week(222222))
+    self.assertFalse(CsvImporter.is_sane_week(202000))
+    self.assertFalse(CsvImporter.is_sane_week(202054))
+    self.assertFalse(CsvImporter.is_sane_week(20200418))
+
+  def test_find_csv_files(self):
+    """Recursively explore and find CSV files."""
+
+    path_prefix = 'prefix/to/the/data/'
+    glob_paths = [
+      # valid weekly
+      path_prefix + 'fb_survey/weekly_202015_county_cli.csv',
+      # valid daily
+      path_prefix + 'ght/20200408_state_rawsearch.csv',
+      # invalid
+      path_prefix + 'invalid/hello_world.csv',
+      # invalid day
+      path_prefix + 'invalid/22222222_b_c.csv',
+      # invalid week
+      path_prefix + 'invalid/weekly_222222_b_c.csv',
+      # invalid geography
+      path_prefix + 'invalid/20200418_country_c.csv',
+      # ignored
+      path_prefix + 'ignored/README.md',
+    ]
+    mock_glob = MagicMock()
+    mock_glob.glob.return_value = glob_paths
+
+    found = list(CsvImporter.find_csv_files(path_prefix, glob=mock_glob))
+
+    expected = [
+      (glob_paths[0], ('fb_survey', 'cli', 'week', 'county', 202015)),
+      (glob_paths[1], ('ght', 'rawsearch', 'day', 'state', 20200408)),
+      (glob_paths[2], None),
+      (glob_paths[3], None),
+      (glob_paths[4], None),
+      (glob_paths[5], None),
+    ]
+    self.assertEqual(found, expected)
+
+  def test_is_header_valid_allows_extra_columns(self):
+    """Allow and ignore extra columns in the header."""
+
+    columns = CsvImporter.REQUIRED_COLUMNS
+
+    self.assertTrue(CsvImporter.is_header_valid(columns))
+    self.assertTrue(CsvImporter.is_header_valid(columns | {'foo', 'bar'}))
+
+  def test_floaty_int(self):
+    """Parse ints that may look like floats."""
+
+    self.assertEqual(CsvImporter.floaty_int('-1'), -1)
+    self.assertEqual(CsvImporter.floaty_int('-1.0'), -1)
+
+    with self.assertRaises(ValueError):
+      CsvImporter.floaty_int('-1.1')
+
+  def test_maybe_apply(self):
+    """Apply a function to a value as long as it's not null-like."""
+
+    self.assertEqual(CsvImporter.maybe_apply(float, '3.14'), 3.14)
+    self.assertEqual(CsvImporter.maybe_apply(int, '1'), 1)
+    self.assertIsNone(CsvImporter.maybe_apply(int, 'NA'))
+    self.assertIsNone(CsvImporter.maybe_apply(int, 'NaN'))
+    self.assertIsNone(CsvImporter.maybe_apply(float, ''))
+    self.assertIsNone(CsvImporter.maybe_apply(float, None))
+
+  def test_extract_and_check_row(self):
+    """Apply various sanity checks to a row of data."""
+
+    def make_row(
+        geo_type='state',
+        geo_id='vi',
+        val='1.23',
+        se='4.56',
+        sample_size='100.5',
+        direction='0'):
+      row = MagicMock(
+          geo_id=geo_id,
+          val=val,
+          se=se,
+          sample_size=sample_size,
+          direction=direction)
+      return geo_type, row
+
+    # cases to test each failure mode
+    failure_cases = [
```

Comment:
maybe also add cases for where non-optional fields are missing?

Reply:
done